src/tallies/TallyBase.C

```diff
@@ -407,6 +407,15 @@ TallyBase::getTallyID() const
   return getWrappedTally()->id();
 }
 
+int
+TallyBase::scoreIndex(const std::string & score) const
+{
+  if (!hasScore(score))
```

Comment:
In Cardinal, we don't typically use coding of return values to indicate something is not found. Would it make more sense to throw an error here instead? If not, can we make the `-1` into a const expr like `INVALID_SCORE` so that it's clearer what the number signifies?